### PR TITLE
BAEL-3075 - Oracle Connection Pooling With Spring

### PR DESCRIPTION
**Pull request by aitorcuesta**
BAEL-3075 - Oracle Connection Pooling With Spring

### CODE REVIEW

File: persistence-modules/spring-boot-persistence-2/README.md [hunk 1]
```diff
@@ -1,3 +1,4 @@
 ### Relevant Articles:
 
 - [Using JDBI with Spring Boot](https://www.baeldung.com/spring-boot-jdbi)
+- [Oracle Connection Pooling With Spring](https://www.baeldung.com/oracle-connection-pooling-with-spring)
```
Comment (eric-martin) [inline persistence-modules/spring-boot-persistence-2/README.md:4]: Don't add your article to the readme. We have a post-processing step that handles this.

File: persistence-modules/spring-boot-persistence-2/pom.xml [hunk 2]
```diff
@@ -41,10 +41,20 @@
     </dependencyManagement>
 
     <dependencies>
-        <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-jdbc</artifactId>
-        </dependency>
+    	<dependency>
```
Comment (eric-martin) [inline persistence-modules/spring-boot-persistence-2/pom.xml:44]: Tabs/spaces

File: persistence-modules/spring-boot-persistence-2/README.md (hunk 1 above)
Comment (aitorcuesta) [reply persistence-modules/spring-boot-persistence-2/README.md]: Removed from readme

File: persistence-modules/spring-boot-persistence-2/pom.xml (hunk 2 above)
Comment (aitorcuesta) [reply persistence-modules/spring-boot-persistence-2/pom.xml]: Changed tabs for spaces

File: persistence-modules/spring-boot-persistence-2/pom.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0"
-         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (eric-martin) [inline persistence-modules/spring-boot-persistence-2/pom.xml:4]: Can you undo this whitespace change?

File: persistence-modules/spring-boot-persistence-2/README.md [hunk 4]
```diff
@@ -1,3 +1,3 @@
 ### Relevant Articles:
 
-- [Using JDBI with Spring Boot](https://www.baeldung.com/spring-boot-jdbi)
+- [Using JDBI with Spring Boot](https://www.baeldung.com/spring-boot-jdbi)
```
Comment (eric-martin) [inline persistence-modules/spring-boot-persistence-2/README.md:3]: Let's remove this file from your PR (no updates needed)

File: persistence-modules/spring-boot-persistence-2/pom.xml [hunk 5]
```diff
@@ -43,7 +43,17 @@
...
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>net.bytebuddy</groupId>
+            <artifactId>byte-buddy</artifactId>
```
Comment (eric-martin) [inline persistence-modules/spring-boot-persistence-2/pom.xml:56]: Is byte-buddy used?

Comment (aitorcuesta) [reply persistence-modules/spring-boot-persistence-2/pom.xml:56]: It's been used at runtime. Let me try to explain to you.
We are using hibernate-core (5.3.11.File). This dependency comes from spring-boot-starter-jpa (2.1.8.RELEASE). So, hibernate-core needs byte-buddy (1.9.16) for running correctly.
We are also using mockito-core (3.3.0), I'm not really sure where this dependency comes from. This mockito-core uses the 1.10.5 version from byte-buddy. So maven is omitting the 1.9.16 version (the one we need), and as the mockito-core is declared as test scope the byte-buddy.jar is not being included. This means that when you try to run the application you get a java.lang.ClassNotFoundException: net.bytebuddy.matcher.ElementMatchers
As I said, this only happens when you use the mvn spring-boot:run command.

If you want, I can remove this declaration, as I said mockito-core is declaring this dependency at test scope. So, all the tests are running fine

File: persistence-modules/spring-boot-persistence-2/pom.xml (hunk 3 above)
Comment (aitorcuesta) [reply persistence-modules/spring-boot-persistence-2/pom.xml]: Done

File: persistence-modules/spring-boot-persistence-2/README.md (hunk 4 above)
Comment (aitorcuesta) [reply persistence-modules/spring-boot-persistence-2/README.md]: I've replaced the README.md with the master version.

**Pull request merged**